src/View/Helper/PaginatorHelper.php

```diff
@@ -1169,4 +1170,41 @@ public function implementedEvents()
     {
         return [];
     }
+
+    /**
+     * dropdown array for Select limit
+     *
+     * @param array $limits This is array of options.
+     * @param int|null $default Default limit for option selecting. Default value is $this->param('perPage').
+     * @param array $options Options for Select tag attributes like class, id or event
+     *
+     * @return string html output.
+     */
+    public function limitControl(array $limits = [], $default = null, array $options = [])
+    {
+        $out = $this->Form->create(null, ['type' => 'get']);
+
+        if (empty($default) || !is_numeric($default)) {
+            $default = $this->param('perPage');
+        }
+
+        if (empty($limits)) {
+            $limits = [
+                '20' => '20',
+                '50' => '50',
+                '100' => '100'
+            ];
+        }
+
+        if (!in_array($default, $limits)) {
```

Comment:
Actually, this have another usage. I am using cache for pagination limit which will stored each page pagination limit value. Now, if anyone wants his own value which is not included in the Limit options, that will be added to the limits. 

Yes, in one sense, this really doesn't make sense here unless anyone using caching or other technique on pagination